flink-table/flink-table-planner/src/main/java/org/apache/flink/table/catalog/DatabaseCalciteSchema.java

```diff
@@ -77,6 +84,23 @@ public Table getTable(String tableName) {
 						!connectorTable.isBatch(),
 						FlinkStatistic.of(tableSource.getTableStats().orElse(null))))
 					.orElseThrow(() -> new TableException("Cannot query a sink only table."));
+			} else if (table instanceof CatalogTable) {
+				Optional<TableFactory> tableFactory = catalog.getTableFactory();
+				TableSource<Row> tableSource = tableFactory.map(tf -> ((TableSourceFactory) tf).createTableSource((CatalogTable) table))
+					.orElse(TableFactoryUtil.findAndCreateTableSource(((CatalogTable) table).toProperties()));
+
+				if (!(tableSource instanceof StreamTableSource)) {
```

Comment:
the message mentions `BoundedTableSource`, so shall we check for it?

Reply:
The message was from Dawid's change. I didn't add it.

Reply:
Actually the message has changed. BoundedTableSource isn't referred any more.